### PR TITLE
fix: provide empty string to 0 length process environment variables

### PR DESCRIPTION
**Pull request by liz3**
### What does this PR do?

For process loaded env vars, its a bug to give them the literal value '""' if the provided length is 0.
I am not entirely sure if this is expected behaviour for DotEnv Files, which is why i left the constant unmodified and opted to only modify the process based value, to which is should apply.

If the current behaviour is also faulty for dotenv, changing the `empty_string_value` might be a better approach

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?
Compiled/ran a test example to verify the behaviour, can add tests if needed.

### CODE REVIEW

File: src/env_loader.zig [hunk 1]
```diff
@@ -361,7 +361,7 @@ pub const Loader = struct {
                     if (value.len > 0) {
                         this.map.put(key, value) catch unreachable;
                     } else {
-                        this.map.put(key, empty_string_value) catch unreachable;
+                        this.map.put(key, "") catch unreachable;
```
Comment (paperclover) [inline src/env_loader.zig]: if it's just going to be an empty string, we no longer need the if branch, as `value` will be an empty string if it's zero length.

also, can you check if `empty_string_value` is used anywhere else, and if not, remove it too

**Review by paperclover — CHANGES_REQUESTED** [review]
thanks for the one line fix, though we can make the code in this function a little simpler if we arent going to use an empty string.

also can you add a test case to `env.test.ts`, probably copying the existing test `.env with zero length strings`... somehow that test already asserts this behavior but in a different case.

File: src/env_loader.zig (hunk 1 above)
Comment (liz3) [reply src/env_loader.zig]: `empty_string_value` is used in quite a few places

**Comment by liz3** [comment]
> thanks for the one line fix, though we can make the code in this function a little simpler if we arent going to use an empty string.
> 
> also can you add a test case to `env.test.ts`, probably copying the existing test `.env with zero length strings`... somehow that test already asserts this behavior but in a different case.

I added a test for the process based case, i want to clarify again that im unsure whether its expected behaviour for DotEnv files to return a quoted string or they too should behave like this.
If thats the case we could change the `empty_string_value` constant to BE a empty string literal rather then only for this case.

**Comment by paperclover** [comment]
> I want to clarify again that im unsure whether its expected behaviour for DotEnv files to return a quoted string or they too should behave like this.

Seems in node.js they will not quote the environment variable.

image

I don't know why we ever did this, tried to find any examples where this is done.

> If thats the case we could change the empty_string_value constant to BE a empty string literal rather then only for this case.

We should just remove that constant. the only usages are done in `if len == 0` checks, besides this following one:

image

for this case let's put `""` inline, which will translate to an empty js string.

**Comment by liz3** [comment]
> > I want to clarify again that im unsure whether its expected behaviour for DotEnv files to return a quoted string or they too should behave like this.
> 
> Seems in node.js they will not quote the environment variable.
> image
> 
> I don't know why we ever did this, tried to find any examples where this is done.
> 
> > If thats the case we could change the empty_string_value constant to BE a empty string literal rather then only for this case.
> 
> We should just remove that constant. the only usages are done in `if len == 0` checks, besides this following one:
> image
> 
> for this case let's put `""` inline, which will translate to an empty js string.

Changes are implemented, can confirm that the behaviour is consistent now!
image

**Review by paperclover — APPROVED** [review]
waiting for tests to run now

**Comment by liz3** [comment]
Format check on `env.test.ts` failed, committed the correction.

**Comment by liz3** [comment]
(the failing of prettier in unrelated to the code now)

**Pull request merged**